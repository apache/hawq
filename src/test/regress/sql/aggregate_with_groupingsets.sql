---
--- Drop existing table
---
DROP TABLE IF EXISTS foo;

---
--- Create new table foo
---
CREATE TABLE foo(type INTEGER, prod VARCHAR, quantity NUMERIC);

---
--- Insert some value
---
INSERT INTO foo VALUES(1, 'Table', 100);
INSERT INTO foo VALUES(2, 'Chair', 250);
INSERT INTO foo VALUES(3, 'Bed', 300);

---
--- Select query with grouping sets
---
SELECT type, prod, sum(quantity) s_quant
FROM
(
  SELECT type, prod, quantity
  FROM foo F1
  LIMIT 3
) F2 GROUP BY GROUPING SETS((type, prod), (prod)) ORDER BY type, s_quant;