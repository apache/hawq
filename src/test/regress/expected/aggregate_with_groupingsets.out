---
--- Drop existing table
---
DROP TABLE IF EXISTS foo;
NOTICE:  table "foo" does not exist, skipping
---
--- Create new table foo
---
CREATE TABLE foo(type INTEGER, prod VARCHAR, quantity NUMERIC);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'type' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
---
--- Insert some value
---
INSERT INTO foo VALUES(1, 'Table', 100);
INSERT INTO foo VALUES(2, 'Chair', 250);
INSERT INTO foo VALUES(3, 'Bed', 300);
---
--- Select query with grouping sets
---
SELECT type, prod, sum(quantity) s_quant
FROM
(
  SELECT type, prod, quantity
  FROM foo F1
  LIMIT 3
) F2 GROUP BY GROUPING SETS((type, prod), (prod)) ORDER BY type, s_quant;
 type | prod  | s_quant 
------+-------+---------
    1 | Table |     100
    2 | Chair |     250
    3 | Bed   |     300
      | Table |     100
      | Chair |     250
      | Bed   |     300
(6 rows)

